iac-extensions/arm/src/test/java/org/sonar/iac/arm/tree/impl/bicep/ResourceDeclarationImplTest.java

```diff
@@ -64,7 +66,7 @@ void shouldParseMinimalResourceDeclarationObject(String body) {
     assertThat(((Identifier) tree.children().get(2)).value()).isEqualTo("myName");
     assertThat(((StringComplete) tree.children().get(3)).value()).isEqualTo("type@version");
     assertThat(((SyntaxToken) tree.children().get(4)).value()).isEqualTo("=");
-    assertThat(tree.children().get(5)).isInstanceOfAny(ObjectExpression.class, IfExpression.class);
+    assertThat(tree.children().get(5)).isInstanceOfAny(ObjectExpression.class, IfExpression.class, ForExpression.class);
```

Comment:
This is not a precise test. It is one of the tests which try to cover everything. When doing a parameterized test you should actually test the differences. The common things can be checked in a dedicated one.
isInstanceOfAny is not precise. 
Additionally, please do not test the elements of a tree with the `children()` method. Aren't there any getters?
Please refactor the entire test approach. I'm happy to discuss a clean approach.

Reply:
I reworked the resource tests.
I discussed there is currently no getter to work with the `ObjectExpression`/`IfCondition`/`ForExpression` yet, so the test mainly focus to ensure we can work with the properties in both cases.